lib/gooddata/lcm/actions/synchronize_ldm.rb

```diff
@@ -35,57 +35,79 @@ class SynchronizeLdm < BaseAction
 
         description 'Allows to have facts with higher precision decimals'
         param :exclude_fact_rule, instance_of(Type::BooleanType), required: false, default: false
+
+        description 'Specifies how to synchronize LDM and resolve possible conflicts'
+        param :synchronize_ldm, instance_of(Type::SynchronizeLDM), required: false, default: 'diff_against_master_with_fallback'
       end
 
       class << self
         def call(params)
-          include_ca = params.include_computed_attributes.to_b
           exclude_fact_rule = params.exclude_fact_rule.to_b
-
+          client = params.gdc_gd_client
           results = []
+          synchronize = params.synchronize.map do |segment_info|
+            sync_segment(client, exclude_fact_rule, include_ca, params, results, segment_info)
+          end
 
-          client = params.gdc_gd_client
-          development_client = params.development_client
+          {
+            results: results,
+            params: {
+              synchronize: synchronize
+            }
+          }
+        end
+
+        private
 
-          synchronize = params.synchronize.map do |info|
-            from_project = info.from
-            to_projects = info.to
+        def sync_segment(client, exclude_fact_rule, params, results, segment_info)
+          from_pid = segment_info[:from]
+          from = params.development_client.projects(from_pid) || fail("Invalid 'from' project specified - '#{from_pid}'")
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)